testing/test_collection.py

```diff
@@ -49,31 +63,33 @@ def test_fail(): assert 0
             assert [1, 2, 3] != fn  # type: ignore[comparison-overlap]
             assert modcol != fn
 
-        assert testdir.collect_by_name(modcol, "doesnotexist") is None
+        assert pytester.collect_by_name(modcol, "doesnotexist") is None
 
-    def test_getparent(self, testdir):
-        modcol = testdir.getmodulecol(
+    def test_getparent(self, pytester: Pytester) -> None:
+        modcol = pytester.getmodulecol(
             """
             class TestClass:
                  def test_foo(self):
                      pass
         """
         )
-        cls = testdir.collect_by_name(modcol, "TestClass")
-        fn = testdir.collect_by_name(testdir.collect_by_name(cls, "()"), "test_foo")
+        cls = pytester.collect_by_name(modcol, "TestClass")
+        fn = pytester.collect_by_name(
+            pytester.collect_by_name(cls, "()"), "test_foo"  # type: ignore[arg-type]
```

Comment:
This `type: ignore` actually indicates a real issue with the typing of `Pytester.collect_by_name` -- the `modcol` argument has type `Module`, but here it is used as a general `Collector`. The implementation seems fine with this. So I suggest we fix it:

1. Change `Pytester.collect_by_name` argument `modcol` type from `Module` to `Collector`.
2. Same for `Testdir.collect_by_name`.
3. Change `Pytester._mod_collections` attribute from `Module` key to `Collector` key.

That should fix the `arg-type` issue.

Reply:
I made this change.  ```arg-type``` issue is gone.